File: Examples/VisualBasic/AsposePDF/Stamps-Watermarks/TextinHeader.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Namespace AsposePDF.StampsWatermarks
    Public Class TextinHeader
        Public Shared Sub Run()
            ' ExStart:TextinHeader
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_StampsWatermarks()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("TextinHeader.pdf"))

            ' Create header
            Dim textStamp As New TextStamp("Header Text")
            ' Set properties of the stamp
            textStamp.TopMargin = 10
            textStamp.HorizontalAlignment = HorizontalAlignment.Center
            textStamp.VerticalAlignment = VerticalAlignment.Top
            ' Add header on all pages
            For Each page As Page In pdfDocument.Pages
                page.AddStamp(textStamp)
            Next

            ' Save updated document
            pdfDocument.Save(dataDir & Convert.ToString("TextinHeader_out_.pdf"))
            ' ExEnd:TextinHeader
            Console.WriteLine(Convert.ToString(vbLf & "Text in header added successfully." & vbLf & "File saved at ") & dataDir)

        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Stamps-Watermarks/TextinHeader.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;

namespace CSharp.AsposePdf.StampsWatermarks
{
    public class TextinHeader
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_StampsWatermarks();

            //open document
            Document pdfDocument = new Document(dataDir+ "TextinHeader.pdf");

            //create header
            TextStamp textStamp = new TextStamp("Header Text");
            //set properties of the stamp
            textStamp.TopMargin = 10;
            textStamp.HorizontalAlignment = HorizontalAlignment.Center;
            textStamp.VerticalAlignment = VerticalAlignment.Top;
            //add header on all pages
            foreach (Page page in pdfDocument.Pages)
            {
                page.AddStamp(textStamp);
            }

            //save updated document
            pdfDocument.Save(dataDir+ "TextinHeader_out.pdf");
 
            
            
        }
    }
}